starfish/agent/invoke_agent.py

```diff
@@ -0,0 +1,60 @@
+"""
+
+Agent class for Invokable Ocean Agents
+
+"""
+
+
+import logging
+import json
+from starfish.account import Account
+from starfish.agent import Agent
+from starfish.listing import Listing
+from starfish.asset import SquidAsset
+from starfish.purchase import Purchase
+from starfish.operation.operation import Operation
+import sys,traceback
+logger = logging.getLogger('ocean')
+import requests
+
+class InvokeAgent:
+    """
+
+    Invoke Agent class allows get a list of operations and invoke them.
+
+    """
+    
+    def __init__(self, **kwargs):
+        """init a standard ocean object"""
+
+        self._koi_url = kwargs.get('koi_url', 'http://localhost:8031/api/v1/brizo/services/')
+
+        r=requests.get(self._koi_url+'operations')
+        j=json.loads(r.text)
+        self.operations=dict(zip([i['name'] for i in j],[i['did'] for i in j]))
```

Comment:
The current tests are failing on this line. I think because you will need to edit travis.yml, to add the correct barge release with koi.
More importantly you need to check to test for JSON and also if the fields `name` and `did` actually exists within the JSON.

I would also if possible move the `requests.get` outside an __init__ to a seperate method, since if this fails, then how do you return an empty object?

Reply:
@billbsing requests.get has been moved out of the constructor, into `get_operations` method

Reply:
w.r.t your comment about build failure, I'll wait for

- this [barge PR ](https://github.com/DEX-Company/barge/pull/11/)to merge
- I'll then add a new tag
- and then change the travis.yml to have the new Barge tag, 
- at which point, hopefully, the tests should pass :)